drivers: charger: Rename BQ25180 driver to BQ2518x and add BQ25186 support

bwitherspoon opened the pull request:
The TI BQ2518x is family of battery charger ICs with compatible configuration registers (BQ25180/BQ25186/BQ25188). This PR renames the existing BQ25180 driver to BQ2518x and adds support for the BQ25186.

bwitherspoon commented on drivers/charger/charger_bq2518x.c:
Its probably rendering weird because it is not just a `git mv`, but any BQ25180/bq25180 strings in the file where also renamed BQ2518X/bq2518x (except for BQ25180_DEVICE_ID). Maybe I could have split that up for easier review.

rriveramcrus replied:
That would be appreciated if you've got the time

bwitherspoon committed:
drivers: charger: Rename BQ25180 driver to BQ2518x

The TI BQ2518x is family of battery charger ICs with compatible
configuration registers (BQ25180/BQ25186/BQ25188). This commit renames
the BQ25180 driver in preparation for adding support for other devices
in this family.

Signed-off-by: Brett Witherspoon <brett@witherspoon.engineering>

bwitherspoon committed:
drivers: charger: bq2518x: Rename internal functions and macros

Rename internal functions and macros BQ25180 to BQ2518x in preparation
for adding support for the BQ25186 device.

Signed-off-by: Brett Witherspoon <brett@witherspoon.engineering>

bwitherspoon committed:
drivers: charger: bq2518x: Add BQ25186 support

Add support for the BQ25186 device to the generic BQ2518x driver.

Signed-off-by: Brett Witherspoon <brett@witherspoon.engineering>

bwitherspoon commented:
Rebased on main and split the renaming commits for easier review. Maybe the last two commits could be squashed.

rriveramcrus commented:
> Maybe the last two commits could be squashed.

@bwitherspoon okay to squash now, thank you for breaking it out for ease of review. Withholding my approval until then to avoid extra notifications, but looks awesome!
